### PR TITLE
Improve INSTALL.md; remove apt update from install-apt.sh

### PR DESCRIPTION
**Pull request by akotlar**
* Remove apt update from install-apt.sh, and instead indicate in installation manual that packages should be updated at the discretion of the user.
* Cleaned up installation instructions

This was motivated by attempting to set up Bystro on a new Ubuntu server. I realized that users will not want Bystro to force package updates, and that the installation of the Python libraries is not spelled out in INSTALL.md

Future PRs will strip out sudo from the install scripts, this is a gentle first step.

### CODE REVIEW

File: install/install-apt-deps.sh [hunk 1]
```diff
@@ -31,4 +30,5 @@ sudo apt install -y nodejs;
 sudo apt install -y npm;
 sudo npm install -g pm2;
 
-sudo apt install -y awscli;
\ No newline at end of file
+sudo apt install -y awscli;
+sudo apt install -y pkg-config;
```
Comment (akotlar) [inline install/install-apt-deps.sh:34]: pkg-config is required for `make build`

File: INSTALL.md [hunk 2]
```diff
@@ -19,29 +17,54 @@ docker run bystro bystro-annotate.pl #Annotate
...
+ 1. The Bystro annotator: a Perl program accessed through the command line (via bin/bystro-*)
+ 2. The Bystro Python package: where the rest of Bystro's functionality lives (statistics, proteomics, etc).
 
-1.  `git clone https://github.com/akotlar/bystro.git && cd bystro && source ./install-rpm.sh`
+## Installing the Bystro annotator (Perl/cli)
+
+##### (Fedora, Redhat, Centos, openSUSE, Mandriva)
 
-## Installation on MacOS
+1.  `git clone https://github.com/akotlar/bystro.git && cd bystro && source ./install-rpm.sh`
```
Comment (poneill) [inline INSTALL.md:30]: here and throughout, do we want to clone from bystrogenomics rather than from your personal branch?

File: INSTALL.md [hunk 3]
```diff
@@ -19,29 +17,54 @@ docker run bystro bystro-annotate.pl #Annotate
...
-###### (tested on HighSierra, interactive)
+##### MacOS (tested on HighSierra, interactive)
 
 1.  `git clone https://github.com/akotlar/bystro.git && cd bystro && source ./install-mac.sh`
 
-## Installation on Debian systems
+##### Ubuntu
+1.  Ensure that packages are up to date (`sudo apt update`), or that you are satisified with the state of package versions.
+2.  `git clone https://github.com/akotlar/bystro.git && cd bystro && source ./install-apt.sh`
+    - Please note that this installation script requires root priveleges, in order to install system dependencies
```
Comment (poneill) [inline INSTALL.md:39]: Maybe more direct to say something like:

`Please not that this installation script will ask you for the root password in order to install system dependencies`

if that's what will happen.

File: install/install-apt-deps.sh (hunk 1 above)
Comment (poneill) [reply install/install-apt-deps.sh:34]: could we just put review comments like these in directly as code comments?

**Review by poneill — DISMISSED** [review]
I think the documentation is heading in a good direction here.

My one blocking concern is whether we need to update the git repos, as it seems like we could easily fall out of synch otherwise-- could we change them or explicitly document that we mean to pull from the personal repo?

File: INSTALL.md (hunk 2 above)
Comment (akotlar) [reply INSTALL.md:30]: Oops good catch

**Comment by akotlar** [comment]
A new build step was created to test setting up our actual dev environment and building the wheel. This is slow, but there's not much we can do about that, besides restricting what files we trigger this on. I did make an attempt at that, but did not yet have success, and it's not clear we want that (because we could easily have dependencies between perl/python, for instance). In any case, the overhead is small, 8 minutes, and is run in parallel with other tests.

**Review by poneill — APPROVED** [review]
lgtm!

**Pull request merged**